src/tokens/TokenDetails.tsx

```diff
@@ -0,0 +1,325 @@
+  }
+
+  if (!token.historicalPricesUsd || !isHistoricalPriceUpdated(token)) {
+    return null
+  }
+
+  return (
+    <View style={styles.priceInfo}>
+      <PercentageIndicator
+        comparedValue={token.historicalPricesUsd.lastDay.price}
+        currentValue={token.priceUsd!}
+        percentageTextStyle={styles.priceInfoText}
+        UpIcon={DataUp}
+        DownIcon={DataDown}
+        NoChangeIcon={DataUp}
+        suffixText={t('tokenDetails.priceDeltaSuffix') ?? undefined}
+        suffixTextStyle={typeScale.bodySmall}
+        testID="TokenDetails/PriceDelta"
+      />
+    </View>
+  )
+}
+
+function Actions({ token }: { token: TokenBalance }) {
+  const { t } = useTranslation()
+  const sendableTokens = useSendableTokens()
+  const swappableTokens = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const cashOutTokens = useCashOutTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const actions = [
+    {
+      name: TokenDetailsActionName.Send,
+      text: t('tokenDetails.actions.send'),
+      iconComponent: QuickActionsSend,
+      onPress: () => {
+        navigate(Screens.Send, { defaultTokenOverride: token.address! })
+      },
+      visible: !!sendableTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId),
+    },
+    {
+      name: TokenDetailsActionName.Swap,
+      text: t('tokenDetails.actions.swap'),
+      iconComponent: QuickActionsSwap,
+      onPress: () => {
+        navigate(Screens.SwapScreenWithBack, { fromTokenId: token.tokenId })
+      },
+      visible:
+        isSwapEnabled &&
+        !!swappableTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId) &&
+        token.balance.gt(TOKEN_MIN_AMOUNT),
+    },
+    {
+      name: TokenDetailsActionName.Add,
+      text: t('tokenDetails.actions.add'),
+      iconComponent: QuickActionsAdd,
+      onPress: () => {
+        navigate(Screens.FiatExchangeAmount, {
+          currency: token.symbol as CiCoCurrency,
+          flow: CICOFlow.CashIn,
+          network: Network.Celo,
+        })
+      },
+      visible: !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId),
+    },
+    {
+      name: TokenDetailsActionName.Withdraw,
+      text: t('tokenDetails.actions.withdraw'),
+      iconComponent: QuickActionsWithdraw,
+      onPress: () => {
+        navigate(Screens.FiatExchangeAmount, {
+          currency: token.symbol as CiCoCurrency,
+          flow: CICOFlow.CashOut,
+          network: Network.Celo,
+        })
+      },
+      visible: !!cashOutTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId),
+    },
+  ].filter((action) => action.visible)
+
+  const moreAction = {
+    name: TokenDetailsActionName.More,
+    text: t('tokenDetails.actions.more'),
+    iconComponent: QuickActionsMore,
+    onPress: () => {
+      // TODO(ACT-917): open bottom sheet
+    },
+  }
+
+  const actionButtons = actions.length > 3 ? [actions[0], actions[1], moreAction] : actions
```

Comment:
nit: magic numbers. probly better to put 3 into `MAX_BUTTONS_PER_ROW`

Reply:
fixed